2016-02-03  Zhongyang Liu  <zhongyan@lnxsrv09.seas.ucla.edu>

	* lib/timespec.h (timespec_cmp):
	Direct comparison to avoid integer overflow with subtraction.
	This avoids problems with ancient times.

--- lib/timespec.h	2016-02-03 19:29:11.924581000 -0800
+++ lib/timespec.h	2016-02-03 19:31:14.719476000 -0800
@@ -45,8 +45,8 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  return a.tv_sec > b.tv_sec ? 1 : (a.tv_sec < b.tv_sec ?
+				    -1 : a.tv_nsec - b.tv_nsec);
 }
 
 # if ! HAVE_DECL_NANOSLEEP
